### PR TITLE
Add 'type-literal-delimiter' rule

### PR DESCRIPTION
**Pull request by andy-hanson**
#### PR checklist

- [ ] Addresses an existing issue: #0000
- [X] New feature, bugfix, or enhancement
  - [X] Includes tests
- [X] Documentation update

#### Overview of change:

Added the `type-literal-delimiter` rule, which prefers `{ x: number, y: string }` over `{ x: number; y: string }`.

#### CHANGELOG.md entry:

[new-rule] `type-literal-delimiter`

### CODE REVIEW

File: test/rules/type-literal-delimiter/test.ts.lint [hunk 1]
```diff
@@ -0,0 +1,28 @@
+type T = {
```
Comment (nchen63) [inline test/rules/type-literal-delimiter/test.ts.lint]: does it handle space before/after delimiter?

Comment (andy-hanson) [inline test/rules/type-literal-delimiter/test.ts.lint]: Not currently, no. I think that would be more appropriate for the `whitespace` rule.

Comment (nchen63) [inline test/rules/type-literal-delimiter/test.ts.lint]: I'm asking if it would erroneously flag { a: b , c: d   }

Comment (andy-hanson) [inline test/rules/type-literal-delimiter/test.ts.lint]: Oh, gotcha. Added a test.

**Comment by adidahiya** [comment]
wait, this feels pretty inconsistent with how we write out interfaces... users are going to be pretty confused when tslint recommends this format:

```ts
type Foo {
  a: number,
  b: number,
}

interface Bar {
  a: number;
  b: number;
}
```

can we please make the default config prefer semicolons?

**Comment by andy-hanson** [comment]
@adidahiya If so, then the current `trailing-comma` wouldn't handle checking for the last token. Maybe `trailing-comma` should be updated to check for type literal delimiters too, regardless of the kind used?

**Comment by adidahiya** [comment]
@andy-hanson no, I think we can leave trailing-comma as-is. I think _this_ rule (type-literal-delimiter) should enforce the trailing semicolon.

**Pull request merged**